### PR TITLE
fix: add definitions for about statuses

### PR DESCRIPTION
**Pull request by davidruvolo51**
As noted in #3423, the action for *about statuses* is not enabled. When the link is clicked, nothing happens. This PR enables this functionality by adding definitions for each status. Definitions are implemented as tooltips (via the floating-vue library). 

## Summary of changes

* Revised legends and added the official statuses and definitions. Definitions are now tooltips.
* Status was added to table cells a `sr-only` element and the title attribute (this provides more context to cells)
* `flex-row-reverse` was dropped from the `<HarmonizationTable>` component where `<Harmonization>` is used

## How to test
- Navigate to the preview
- Navigate to `/catalogue-demo/ssr-catalogue/all/variables?page=1&view=harmonization`
- Note the legend at the top of the table and click *About statuses* to view the definitions
- Pick a variable with data available and navigate to the variable details view.

## Todo
- [x] Official definitions needed
- [x] Confirmation on modal content: title and description
- [x] Create variations of the harmonisation legend for both views
- [ ] Keep `No data` status? If so, how is this defined?
- ~~Recalculate statuses~~ (will add in another PR)

### CODE REVIEW

File: apps/nuxt3-ssr/components/harmonization/HarmonizationLegend.vue [hunk 1]
```diff
@@ -7,27 +7,75 @@ const props = withDefaults(
...
+        <dd>
+          Harmonization is incomplete. It may be the cases that this process is
+          ongoing or cannot progress further.
+        </dd>
+        <dt>No Data</dt>
+        <dd>
+          Harmonization is unknown. Either information is not available or it is
+          not possible to harmonize these variables.
+        </dd>
+      </dl>
```
[the remark this answers is not in the document]
Comment (davidruvolo51) [reply apps/nuxt3-ssr/components/harmonization/HarmonizationLegend.vue:78]: I created new variations of the legend component: one for the detailed matrix view and another for the harmonisation view.

**Comment by connoratrug** [comment]
Could we keep the design, it feels more consistent with the rest of the application, in general i remember icon only performs worse then icon+text or text. 

design:
Screenshot 2024-03-13 at 09 03 26

current imp:

Screenshot 2024-03-13 at 09 00 01

pr:

Screenshot 2024-03-13 at 09 07 21

File: apps/nuxt3-ssr/components/harmonization/HarmonizationLegend.vue [hunk 2]
```diff
@@ -7,27 +7,75 @@ const props = withDefaults(
...
+        <dt>Partial</dt>
+        <dd>
+          Harmonization is incomplete. It may be the cases that this process is
+          ongoing or cannot progress further.
+        </dd>
+        <dt>No Data</dt>
+        <dd>
+          Harmonization is unknown. Either information is not available or it is
+          not possible to harmonize these variables.
+        </dd>
```
Comment (connoratrug) [inline apps/nuxt3-ssr/components/harmonization/HarmonizationLegend.vue:77]: bit confused , we dont need the modal any more now that we use the popover , or am i missing something here and looking at the wrong 
Screenshot 2024-03-13 at 12 54 02

**Review by connoratrug — CHANGES_REQUESTED** [review]
- Not all of the possible states are shown in the legend now ( what does the white square mean with no icon) 
- The same color code are used used for different states ( Available, Completed)
- The original design included a separate 'link' what suggested to a information on all of the possible status would be shown together , this we safe to use from having to hover over each state individually.
- There is no visual indication that the popup info is available 

In short i really do prefer the original design ( as implemented ) and would suggest jsut adding the popover/modal with status details when the 'About statuses' links is clicked or hovered over.

**Comment by davidruvolo51** [comment]
I reverted to the original design. There are now two legend components: 1) [Harmonisation Matrix](https://github.com/molgenis/molgenis-emx2/blob/fix/about-status/apps/nuxt3-ssr/components/harmonization/HarmonizationLegendMatrix.vue) and 2) [Detailed Matrix Legend](https://github.com/molgenis/molgenis-emx2/blob/fix/about-status/apps/nuxt3-ssr/components/harmonization/HarmonizationLegendDetailed.vue). The original [Harmonization Legend](https://github.com/molgenis/molgenis-emx2/blob/fix/about-status/apps/nuxt3-ssr/components/harmonization/HarmonizationLegend.vue) is still present as I'm not sure if it is used elsewhere. Please advise if this component should be removed. 

In the [Harmonisation Matrix](https://github.com/molgenis/molgenis-emx2/blob/fix/about-status/apps/nuxt3-ssr/components/harmonization/HarmonizationLegendMatrix.vue) component, there is only one term and definition to display. (as listed in the original issue). To distinguish themes between legends, the `available` status is styled using `bg-blue-500` and `fill-white`. Instead of rewritting the harmonisation status mapping, I added a temporary work around by adding the `available` to the [Harmonization Types](https://github.com/molgenis/molgenis-emx2/blob/c1d417cdf73f8f240abc4899670ab0e8b542bce4/apps/nuxt3-ssr/interfaces/types.ts#L262). (This will be addressed in the following PR.)

Styling is unchanged in the [Detailed Matrix Legend](https://github.com/molgenis/molgenis-emx2/blob/fix/about-status/apps/nuxt3-ssr/components/harmonization/HarmonizationLegendDetailed.vue), but there are new terms and definitions were added as a tooltip on the "about status" text element.

I opted for the `span` element for the "about status" label as a button or anchor element wouldn't make sense as there isn't a redirect or action associated with this element (besides triggering the tooltip).

The status for `No data` wasn't included in the official list. If this needed, I can revert this deletion.

File: apps/nuxt3-ssr/components/harmonization/HarmonizationLegend.vue (hunk 2 above)
Comment (davidruvolo51) [reply apps/nuxt3-ssr/components/harmonization/HarmonizationLegend.vue:77]: See comment below: https://github.com/molgenis/molgenis-emx2/pull/3498#issuecomment-1995484442

File: apps/nuxt3-ssr/components/harmonization/HarmonizationTableCellAvailableIcon.vue [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+const recodedStatus = computed(() => {
+  if (availableStatuses.includes(props.status)) {
+    return "available";
+  }
+  return props.status;
+});
+</script>
+<template>
+  <div
+    :data-status="recodedStatus"
```
Comment (connoratrug) [inline apps/nuxt3-ssr/components/harmonization/HarmonizationTableCellAvailableIcon.vue:19]: whats the dataStatus ?

File: apps/nuxt3-ssr/components/harmonization/HarmonizationTableCellAvailableIcon.vue [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  status: HarmonizationStatus;
+}>();
+
+const availableStatuses = ["partial", "complete"];
+
+const recodedStatus = computed(() => {
+  if (availableStatuses.includes(props.status)) {
+    return "available";
+  }
+  return props.status;
```
Comment (connoratrug) [inline apps/nuxt3-ssr/components/harmonization/HarmonizationTableCellAvailableIcon.vue:14]: this feels somewhat indirect , why not pass 'available' as status instead of this computation at the Icon level

**Review by connoratrug — APPROVED** [review]
Looks oke me;

- Some small code suggestions about the status computation ( see code comment)
- Please add a e2e test for both table and the single variable detail 
- The on hover style for the grid cell is missing 
 Screenshot 2024-03-18 at 13 11 06

- Might be beter 
to render the legend status is vertically for small-screen modal view 
Screenshot 2024-03-18 at 13 10 53

File: apps/nuxt3-ssr/components/harmonization/HarmonizationTableCellAvailableIcon.vue (hunk 3 above)
Comment (davidruvolo51) [reply apps/nuxt3-ssr/components/harmonization/HarmonizationTableCellAvailableIcon.vue:19]: This line was added to eliminate the computed values for css classes. This line creates a `data-*` attribute and binds the `status` to it. The icon is then styled using data attribute selector rather computed variables.

File: apps/nuxt3-ssr/components/harmonization/HarmonizationTableCellAvailableIcon.vue (hunk 4 above)
Comment (davidruvolo51) [reply apps/nuxt3-ssr/components/harmonization/HarmonizationTableCellAvailableIcon.vue:14]: Removed

**Comment by connoratrug** [comment]
i think most if the stuff in there now , suggest we merge this pr and we can add e2e tests and finish tweaking this style in separate PR

**Comment by davidruvolo51** [comment]
Closes #3423
